### PR TITLE
Fixing out of range crash

### PR DESCRIPTION
**Pull request by zackcam**
Fix for: https://github.com/valkey-io/valkey-json/issues/105

This fix uses the suggested change of switching to strtod. As this will convert overflows to inf, and not just throw an error. This library still works on subnormal values as well. We do not need an extra check for ERANGE from what I could tell as we are covered by the inf check in `JSONUTIL_ADDITION_OVERFLOW`

Two new tests are created the one in dom checks for three different cases, normal + subnormal, subnormal + normal then an overflow check.

### CODE REVIEW

File: tst/unit/dom_test.cc [hunk 1]
```diff
@@ -777,6 +777,44 @@ TEST_F(DomTest, testNumIncrMultBy_string_value_overflow) {
...
+
+    // Case 1: Stored value is normal, the increment 1e-308 is subnormal and is converted back via
+    // GetDouble() during the addition.
+    JsonUtilCode rc = dom_set_value(nullptr, doc1, ".foo", "1.5", false, false);
+    EXPECT_EQ(rc, JSONUTIL_SUCCESS);
+    rc = dom_increment_by(doc1, ".foo", &parser.Parse("1e-308", 6).GetJValue(), res, isV2Path);
+    EXPECT_EQ(rc, JSONUTIL_SUCCESS);
+    EXPECT_EQ(res.size(), 1);
+    EXPECT_EQ(res[0], 1.5 + 1e-308);
+    EXPECT_FALSE(isV2Path);
```
Comment (roshkhatri) [inline tst/unit/dom_test.cc:797]: I think here we should be testing subnormal+subnormal to see that the conversion is happening  something like 1e-308 + 2e-308. Now the answer ≈3e-308, so now the answer genuinely depends on both numbers being converted correctly.

This test doesn't actually prove the tiny number was handled correctly. It only proves "it didn't crash."

WDYT?

**Review by roshkhatri — COMMENTED** [review]
The fix looks great, just a comment about the test case

also can we add a test for a huge negative number, which would produce negative infinity.

File: tst/unit/selector_test.cc [hunk 2]
```diff
@@ -1289,6 +1289,25 @@ TEST_F(SelectorTest, test_delete) {
...
+    JsonUtilCode rc = dom_parse(nullptr, input, strlen(input), &d1);
+    EXPECT_EQ(rc, JSONUTIL_SUCCESS);
+
+    // Build the plain-digit literal 10^309, which is > DBL_MAX.
+    std::string big_literal = "1" + std::string(309, '0');
+    std::string path = "$[?(@.n > " + big_literal + ")]";
+
+    Selector selector;
+    rc = selector.getValues(*d1, path.c_str());
+    // 5 is not greater than 10^309, so no values match and it doesnt crash.
```
Comment (roshkhatri) [inline tst/unit/selector_test.cc]:
```suggestion
    // 5 is not greater than 10^309, so no values match and it doesn't crash.
```

**Review by roshkhatri — APPROVED** [review]
I think this looks good, the tests we have are the failure cases in the issue so that okay
